test/Grains/TestGrainInterfaces/IValueTypeTestGrain.cs

```diff
@@ -144,8 +144,16 @@ public interface IRoundtripSerializationGrain : IGrainWithIntegerKey
         Task<CampaignEnemyTestType> GetEnemyType();
 
         Task<object> GetClosedGenericValue();
+
+        Task<RetVal> GetRetValForParamVal(ParamVal param);
     }
 
+    [Serializable]
+    public record ParamVal(int Value);
```

Comment:
@ReubenBond Does this mean records have to have the Serializable attribute?

Reply:
The same rules apply for records as for other types: they don't need `[Serializable]` if they directly appear in another serializable type or a grain interface method signature. I'd recommend adding `[Serializable]` to any serializable type, though.